src/lib.rs

```diff
@@ -1146,18 +1148,87 @@ pub unsafe trait NoCell {
         Self: Sized;
 }
 
-/// Types whose validity can be checked at runtime, allowing them to be
-/// conditionally converted from byte slices.
+/// Implements [`TryFromBytes`].
 ///
-/// WARNING: Do not implement this trait yourself! Instead, use
-/// `#[derive(TryFromBytes)]`.
+/// This derive synthesizes the runtime checks required to check if a sequence
+/// of initialized bytes corresponds to a valid instance of a type. This derive
+/// can be applied to structs, enums, and unions; e.g.:
 ///
-/// `TryFromBytes` types can safely be deserialized from an untrusted sequence
-/// of bytes by performing a runtime check that the byte sequence contains a
-/// valid instance of `Self`.
+/// ```
+/// # use zerocopy_derive::{TryFromBytes, NoCell};
+/// #[derive(TryFromBytes)]
+/// struct MyStruct {
+/// # /*
+///     ...
+/// # */
+/// }
 ///
-/// `TryFromBytes` is ignorant of byte order. For byte order-aware types, see
-/// the [`byteorder`] module.
+/// #[derive(TryFromBytes)]
+/// #[repr(u8)]
+/// enum MyEnum {
+/// #   V00,
+/// # /*
+///     ...
+/// # */
+/// }
+///
+/// #[derive(TryFromBytes, NoCell)]
+/// union MyUnion {
+/// #   variant: u8,
+/// # /*
+///     ...
+/// # */
+/// }
+/// ```
+///
+/// [safety conditions]: trait@TryFromBytes#safety
+#[cfg(any(feature = "derive", test))]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "derive")))]
+pub use zerocopy_derive::TryFromBytes;
+
+/// Types for which some bit patterns are valid.
+///
+/// A memory region of the appropriate length which contains initialized bytes
+/// can be viewed as a `TryFromBytes` type so long as the runtime value of bytes
+/// correspond to a [*valid instance*] of that type. For example, [`bool`] is
+/// `TryFromBytes`; zerocopy can transmute a [`bool`] into a [`u8`] so long as
```

Comment:
```suggestion
/// `TryFromBytes`; zerocopy can transmute a [`u8`] into a  [`bool`] so long as
```

Also, this isn't technically true in our API (right now). We don't have `try_transmute!`, so we can't actually do this. Maybe say something like "zerocopy can transmute a `[u8]` into a `bool`..."?

Reply:
Done